pkg/execution/realtime/api_token_auth.go

```diff
@@ -0,0 +1,52 @@
+package realtime
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/inngest/inngest/pkg/api/apiv1/apiv1auth"
+)
+
+type claimsKeyTyp string
+
+const claimsKey = claimsKeyTyp("rt-claims")
+
+// realtimeAuthMW attempts to auth via realtime JWTs, falling back to the original auth
+// middleware if no JWT was found.
+func realtimeAuthMW(jwtSecret []byte, mw func(http.Handler) http.Handler) func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			ctx := r.Context()
+
+			// Check to see if a valid realtime JWT is being sent as a bearer token.
+			key := r.Header.Get("Authorization")
+			if key != "" && len(key) > 8 {
+				// Remove "Bearer " prefix
+				key = key[7:]
+			}
+
+			claims, err := ValidateJWT(r.Context(), jwtSecret, key)
+			if err == nil {
+				// Update the request's context with the claims
+				r = r.WithContext(context.WithValue(ctx, claimsKey, claims))
+				// We have a valid set of claims
+				next.ServeHTTP(w, r)
+				return
+			}
+
+			// Call the original middelware.
+			if mw != nil {
+				mw(next)
+				return
+			}
+			next.ServeHTTP(w, r)
+		})
+	}
+}
+
+func realtimeAuth(ctx context.Context, af apiv1auth.AuthFinder) (apiv1auth.V1Auth, error) {
```

Comment:
How does cloud read the claims from the context when we implement the AuthFinder? Right now, this isn't exposed

Reply:
It creates a new copy of the API which does this automatically.